hermeto/core/models/input.py

```diff
@@ -21,11 +31,33 @@
 ModelT = TypeVar("ModelT", bound=pydantic.BaseModel)
 
 
+def _is_experimental_package_error(error: "ErrorDict") -> Optional[tuple[str, str]]:
+    """Check if error is for experimental package and return (base_name, x_name) if so."""
+    if error.get("type") != "union_tag_invalid":
+        return None
+
+    tag = error.get("ctx", {}).get("tag", "")
+    x_prefixed = f"x-{tag}"
+
+    if x_prefixed in get_args(PackageManagerType):
+        return tag, x_prefixed
+
+    return None
+
+
 def parse_user_input(to_model: Callable[[T], ModelT], input_obj: T) -> ModelT:
     """Parse user input into a model, re-raise validation errors as InvalidInput."""
     try:
         return to_model(input_obj)
     except pydantic.ValidationError as e:
+        experimental_errors = [_is_experimental_package_error(err) for err in e.errors()]
+        experimental_suggestions = [exp for exp in experimental_errors if exp is not None]
+
+        if experimental_suggestions:
+            unsupported = [base for base, _ in experimental_suggestions]
+            suggested = [x_name for _, x_name in experimental_suggestions]
```

Comment:
This is a good use of list comprehensions! Optional suggestion: you can get even further with a `zip`:
```python
unsupported, suggested = zip(*experimental_suggestions)
```